auth-server/src/middleware/requireLogin.ts

```diff
@@ -32,7 +40,33 @@ const requireLogin = (req: Request, res: Response, next: () => void) => {
     return res.status(401).json({ error: 'access denied' });
   }
 
+  req.decodedToken = decodedToken;
+
   next();
 };
 
-export default requireLogin;
+export const requireBeaconToken = (
+  req: Request & DecodedToken,
+  res: Response,
+  next: () => void
+) => {
+  const token = getTokenFrom(req);
+  if (!token) {
+    return res.status(401).json({ error: 'token missing' });
+  }
+
+  let decodedToken;
+  try {
+    decodedToken = jwt.verify(token, appConfig.JWT_SECRET) as any;
+  } catch (exception) {
+    return res.status(401).json({ error: 'invalid token' });
+  }
+
+  if (!decodedToken.beaconId || decodedToken.beaconId.length === 0) {
+    return res.status(401).json({ error: 'invalid token' });
```

Comment:
Please give more informative error messages for the api user. Something which indicates why the error happened.

Reply:
We have two similar errors with same explanation, it makes it hard to see quickly what is the reason for 401 error.